Fix the config/tokens endpoint

stuartlangridge opened the pull request:
Fix the config/tokens endpoint, which does not work from bin/cc as it currently stands.

bin/cc sends a list of tokens as a simple JSON string, colon-separated, by PUT. This gets rejected by strict body-parser JSON parsing (which expects JSON to be an array or an object), and also by config.js which expects config/tokens to be addressed as POST.
Add a GET verb to the /config/tokens endpoint to list the tokens that have already been added.

(Fixes https://github.com/Microsoft/ghcrawler-cli/issues/8)

msftclas commented:
@__stuartlangridge__,
Thanks for your contribution.
To ensure that the project team has proper rights to use your work, please complete the Contribution License Agreement at https://cla.microsoft.com.

It will cover your contributions to all Microsoft-managed open source projects.
_Thanks,
Microsoft Pull Request Bot_

msftclas commented:
__@stuartlangridge__, thanks for signing the contribution license agreement. We will now validate the agreement and then the pull request.
<br />_Thanks, Microsoft Pull Request Bot_

stuartlangridge commented:
Fair point on the exposure. The scenario where it's needed is this: me asking "have I added this token? which ones have I added already?". Not being able to find this out at all is quite frustrating, especially when you're experimenting with the token functionality :) If you think it's too risky then dropping it from the REST API is reasonable, but having _some_ way to list the tokens that the crawler already knows about would be much appreciated!

jeffmcaffer commented:
got it. What we do in the logs is output part of the token (see https://github.com/Microsoft/ghcrawler/blob/develop/providers/fetcher/githubFetcher.js#L51). That is generally enough for debugging purposes.  Would that work for you?

stuartlangridge commented:
It would! That's a good thought, indeed.

stuartlangridge commented:
(pushed a new commit which ellipsises tokens to 8 characters)

jeffmcaffer commented:
Thanks for the changes @stuartlangridge
